### PR TITLE
Clarify that Tracer.extract can return null

### PR DESCRIPTION
**Pull request by mabn**
According to https://github.com/opentracing/opentracing-java/issues/168 in practice the tracer implementations return null when `SpanContext` is not found in the carrier. This change documents this behaviour.

It also adds a dependency on `com.google.code.findbugs:jsr305` but it's optional, it is not listed in the pom and client applications do not need to depend on jsr305. 

Alt text

Compiling the code above with:
```
javac -Werror -Xlint:all -cp \
/Users/mabn/.m2/repository/io/opentracing/opentracing-api/0.30.1-SNAPSHOT/opentracing-api-0.30.1-SNAPSHOT.jar:\
/Users/mabn/.m2/repository/io/opentracing/opentracing-noop/0.30.1-SNAPSHOT/opentracing-noop-0.30.1-SNAPSHOT.jar:\
/Users/mabn/.m2/repository/io/opentracing/opentracing-util/0.30.1-SNAPSHOT/opentracing-util-0.30.1-SNAPSHOT.jar \
Main.java
```
procudes no warnings.

### CODE REVIEW

File: opentracing-api/pom.xml [hunk 1]
```diff
@@ -31,6 +31,14 @@
         <main.basedir>${project.basedir}/..</main.basedir>
         <main.java.version>1.6</main.java.version>
         <main.signature.artifact>java16</main.signature.artifact>
+        <findbugs.jsr305.version>3.0.2</findbugs.jsr305.version>
     </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
```
Comment (pavolloffay) [inline opentracing-api/pom.xml]: Can we do this in a separate PR? It might apply also at different places

Comment (mabn) [reply opentracing-api/pom.xml]: Do you mean to split adding jsr305 and marking `Tracer.extract` into two separate PRs?

**Comment by mabn** [comment]
I'll close this PR as I haven't looked into this for a while.

**Comment by whiskeysierra** [comment]
I'd still like to see this happening, tbh.
